tests/noto_fonts_for_android_test.py

```diff
@@ -121,5 +132,25 @@ def test_font_weights():
   assert not errors, ", ".join(errors)
 
 
+def test_font_full_weight_coverage():
+  root = etree.parse(str(_noto_4_android_file()))
+  errors = []
+  for family in root.iter("family"):
+    font_to_xml_weights = collections.defaultdict(set)
+    for font in family.xpath("//font[@path]"): 
+      font_to_xml_weights[_font_path(font)].add(int(font.attrib["weight"]))
+
+  # now you have a map of font path => set of weights in xml
+  for font_path, xml_weights in font_to_xml_weights.items():
```

Comment:
once you have index in the map key you can capture it here to pass in to _open_font_path

`for (font_path, font_number), xml_weights in font_to_xml_weights.items():`